src/hazelweb/gui/UHCode.re

```diff
@@ -86,25 +73,14 @@ let view =
 
         | Annot(Step(_) | EmptyLine | SpaceOp, l) => go(l)
 
-        | Annot(Token({shape, len, has_cursor}), l) => {
```

Comment:
We can now remove these fields from `Token` and the `find_and_decorate_caret` function from `UHLayout`.

Reply:
CursorMap uses the has_cursor and len fields.

Reply:
oh yeah, never mind